skimage/morphology/misc.py

```diff
@@ -235,3 +243,208 @@ def remove_small_holes(ar, area_threshold=64, connectivity=1, *, out=None):
     np.logical_not(out, out=out)
 
     return out
+
+
+def remove_near_objects(
+    label_image,
+    min_distance,
+    *,
+    priority=None,
+    p_norm=2,
+    spacing=None,
+    out=None,
+):
+    """Remove objects until a minimal distance is ensured.
+
+    Iterates over all objects (pixels that aren't zero) inside an image and
+    removes "nearby" objects until all remaining ones are spaced more than a
+    given minimal distance from each other.
+
+    Parameters
+    ----------
+    label_image : ndarray of integers
+        An n-dimensional array containing object labels, e.g. as returned by
+        :func:`~.label`. A value of zero is considered background, all other
+        object IDs must be positive integers.
+    min_distance : int or float
+        Remove objects with lower priority whose distance is not greater than
```

Comment:
Priority is undefined at this point; try and avoid forward references, unless you make them explicit.

Reply:
I reworded this. Hope that's what you had in mind.